core/domain/state_domain.py

```diff
@@ -916,8 +940,917 @@ def get_all_outcomes(self) -> List[Outcome]:
+        rule_spec_till_now: List[RuleSpecDict] = []
+
+        # Minimum number of selections should be no greater than maximum
+        # number of selections.
+        if min_value > max_value:
+            raise utils.ValidationError(
+                f'Min value which is {str(min_value)} '
+                f'is greater than max value '
+                f'which is {str(max_value)} '
+                f'in ItemSelectionInput interaction.'
+            )
+        choices = self.customization_args['choices'].value
+
+        # There should be enough choices to have minimum number
+        # of selections.
+        if len(choices) < min_value:
+            raise utils.ValidationError(
+                f'Number of choices which is {str(len(choices))} '
+                f'is lesser than the '
+                f'min value selection which is {str(min_value)} '
+                f'in ItemSelectionInput interaction.'
+            )
+        for ans_group_index, answer_group in enumerate(self.answer_groups):
+            for rule_spec_index, rule_spec in enumerate(
+                answer_group.rule_specs
+            ):
+                # Rule should not be duplicate.
+                if rule_spec.to_dict() in rule_spec_till_now:
+                    raise utils.ValidationError(
+                        f'The rule {rule_spec_index} of answer group '
+                        f'{ans_group_index} of ItemSelectionInput interaction '
+                        f'is already present.'
+                    )
+                rule_spec_till_now.append(rule_spec.to_dict())
+
+                # `Equals` should have between min and max number of selections.
+                if rule_spec.rule_type == 'Equals':
+                    if (
+                        len(rule_spec.inputs['x']) < min_value or
+                        len(rule_spec.inputs['x']) > max_value
+                    ):
+                        raise utils.ValidationError(
+                            f'Selected choices of rule \'{rule_spec_index}\' '
+                            f'of answer group \'{ans_group_index}\' '
+                            f'either less than min_selection_value '
+                            f'or greater than max_selection_value '
+                            f'in ItemSelectionInput interaction.'
+                        )
+
+        # All choices should be unique and non-empty.
+        seen_choices = []
+        for choice in choices:
+            if choice.html in ('<p></p>', ''):
+                raise utils.ValidationError(
+                    'There should not be any empty '
+                    'choices in ItemSelectionInput interaction.'
+                )
+            if choice.html in seen_choices:
+                raise utils.ValidationError(
+                    'There should not be any duplicate '
+                    'choices in ItemSelectionInput interaction.'
+                )
+            seen_choices.append(choice.html)
```

Comment:
This still seems to be almost the same as in `_validate_multi_choice_input`

Reply:
Updated